src/www/vpn_ipsec_pubkeys.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/*
+ * Copyright (C) 2019 Pascal Mathis <mail@pascalmathis.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+require_once("guiconfig.inc");
+require_once("filter.inc");
+require_once("services.inc");
+require_once("interfaces.inc");
+require_once("plugins.inc.d/ipsec.inc");
+require_once('phpseclib/Math/BigInteger.php');
+require_once('phpseclib/Crypt/Hash.php');
+require_once('phpseclib/Crypt/RSA.php');
+
+config_read_array('ipsec', 'publickey');
+ipsec_publickey_sort();
+
+function parse_rsa_key($key_str)
+{
+    // Skip parsing of empty key
+    if (!$key_str) {
+        return false;
+    }
+
+    // Attempt loading key as RSA key
+    $rsa_key = new \phpseclib\Crypt\RSA();
```

Comment:
Doesn’t this need require headers to avoid breakage later on?

Reply:
The required includes for \phpseclib\Crypt\RSA() are included earlier in the same file (see line 33 - 36) and the code was tested on a OPNsense machine running the latest release with make mount. phpseclib itself is already a dependency of the existing source code, specifically in system_certmanager.php. I made sure to only include files which are actually required and various public and private keys worked fine with the current list of includes.